baybe/constraints/continuous.py

```diff
@@ -115,25 +130,61 @@ def to_botorch(
         Args:
             parameters: The parameter objects of the continuous space.
             idx_offset: Offset to the provided parameter indices.
+            batch_size: The batch size used for the recommendation. Necessary
+                for interpoint constraints as indices need to be adjusted.
+                Ignored by all other constraints.
 
         Returns:
             The tuple required by botorch.
+
+        Raises:
+            RuntimeError: When the constraint is an interpoint constraint but
+                ``batch_size`` is ``None``.
         """
         import torch
 
         from baybe.utils.torch import DTypeFloatTorch
 
+        # NOTE: The interpoint constraint case requires indices to be a 2-d tensor.
+        # This requires some adjustments to the indices.
+        # See https://botorch.readthedocs.io/en/latest/optim.html, in particular the
+        # docstring of ``optimize_acqf`` for details.
+
         param_names = [p.name for p in parameters]
-        param_indices = [
-            param_names.index(p) + idx_offset
-            for p in self.parameters
-            if p in param_names
-        ]
+        coefficients: list[float]
+        torch_indices: Tensor
+        if not self.is_interpoint:
+            param_indices = [
+                param_names.index(p) + idx_offset
+                for p in self.parameters
+                if p in param_names
+            ]
+            coefficients = self.coefficients
+            torch_indices = torch.tensor(param_indices)
+        elif batch_size is None:
+            raise RuntimeError(
+                "No ``batch_size`` set but using interpoint constraints."
+                "This should not happen and means that there is a bug in the code."
+            )
+        else:
+            param_index_dict = {
+                name: param_names.index(name) for name in self.parameters
+            }
+            param_indices_interpoint = [
+                (batch, param_index_dict[name] + idx_offset)
+                for name in self.parameters
+                for batch in range(batch_size)
+            ]
+            # Collect the coefficients such that there they have the form
+            # [coef_0,...,coef_0, coef_1,...,coef_1,...,coef_n,...,coef_n]
+            # where each coefficient is repeated ``batch_size`` times.
+            coefficients = list(chain(*zip(*repeat(self.coefficients, batch_size))))
```

Comment:
I'm not even trying to parse this construction, which seems to come directly from hell 🔥 😄 

Reading your comment, I'm wondering: instead of having a super complicated expression effectively doing array arithmetic, later converting it to torch, and then adding a three-line comment explaining it, why don't you simply construct it in simple ways using torch tensor operations directly?